craft_parts/overlays/errors.py

```diff
@@ -0,0 +1,53 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Overlay error definitions."""
+
+from craft_parts import errors
+
+
+class OverlayError(errors.PartsError):
+    """Base class for overlay handler errors."""
+
```

Comment:
How about moving most of the logic in Overlay{Unm|M}ountError to the parent?

```suggestion

    def __init__(self, action: str, mountpoint: str, message: str):
        self.action = action
        self.mountpoint = mountpoint
        self.message = message
        
        brief = f"Failed to {action} overly at {mountpoint}: {message}"
        
        super().__init__(brief=brief)
```

Reply:
this may mean you don't need two distinct errors as .action would hold the operation (.action is not a final name, could be .operation :-))

Reply:
I have the feeling that this type of construct is very close to creating a generic exception only containing a message, given that something failing and the failure happening in overlay are implied if this is an overlay error. Since the general pattern is to have more specific exceptions, I would prefer to have separate mount/unmount errors in this case.